docs/developer/architecture/coding-principles.md

```diff
@@ -27,26 +27,64 @@
 
 1. Avoid layers of indirection when they are not needed (e.g. "pass-through methods").
 
-### IV. Testing
+### IV. General Coding Style
+
+1. use `var` instead of explicit types (helps with clarity)
+2. avoid `final` in method args and local variables
+3. use `final` in field declarations
+4. avoid `static` fields except in constants or when absolutely necessary. (you should be able to provide a reason).
+5. use `interfaces` as containers for constants
+6. use "minimally required types" (or "smallest possible API"), e.g. use `ObjectMapper` instead of `TypeManager`
+   , or use a `String` instead of a more complex object containing the String, etc.
```

Comment:
These are probably two separate issues. The string issue is "Don't create wrapper types for Strings and other types that may be passed and referenced explicitly"